nuxeo-features/nuxeo-platform-notification/nuxeo-platform-notification-core/src/main/java/org/nuxeo/ecm/platform/ec/notification/email/EmailHelper.java

```diff
@@ -185,38 +184,30 @@ protected void sendmail0(Map<String, Object> mail)
      * Gets the session from the JNDI.
      */
     private static Session getSession() {
-        Session session = null;
-        if (javaMailNotAvailable) {
-            return null;
-        }
-        // First, try to get the session from JNDI, as would be done under J2EE.
-        try {
-            NotificationService service = (NotificationService) Framework.getRuntime()
-                                                                         .getComponent(NotificationService.NAME);
-            InitialContext ic = new InitialContext();
-            session = (Session) ic.lookup(service.getMailSessionJndiName());
-        } catch (NamingException ex) {
-            log.warn("Unable to find Java mail API", ex);
-            javaMailNotAvailable = true;
+        if (!javaMailNotAvailable) {
+            // First, try to get the session from JNDI, as would be done under J2EE.
+            try {
+                NotificationService service = (NotificationService) Framework.getRuntime()
+                                                                             .getComponent(NotificationService.NAME);
+                return MailSessionBuilder.fromJndi(service.getMailSessionJndiName()).build();
+            } catch (NuxeoException ex) {
+                log.warn("Unable to find Java mail API", ex);
+                javaMailNotAvailable = true;
+            }
         }
-
-        return session;
+        return null;
     }
 
     /**
      * Instantiates a new session that authenticates given the protocol's properties. Initializes also the default
```

Comment:
_Instantiates a new session that authenticates given the protocol's properties_ -> Instantiates a new session that authenticates the given protocol's properties...

Reply:
As seen on slack, we won't do this change as it's on a deprecated method.